src/tracing/exporters/datadog.js

```diff
@@ -157,17 +157,10 @@ class DatadogTraceExporter extends BaseTraceExporter {
 		sc._traceId = this.convertID(span.traceID);
 		sc._spanId = this.convertID(span.id);
 
-		// Activate span in Datadog tracer
-		const asyncId = asyncHooks.executionAsyncId();
-		this.ddScope._spans = this.ddScope._spans || {};
-		const oldSpan = this.ddScope._spans[asyncId];
-
-		this.ddScope._spans[asyncId] = ddSpan;
+		this.ddTracer.scope()._enter(ddSpan);
 
 		span.meta.datadog = {
```

Comment:
Any idea on if this meta property is considered internal? I could see the `asyncId` being useful in some middlewares